src/WopiHost.Core/Controllers/ContainersController.cs

```diff
@@ -141,16 +126,140 @@ public async Task<IActionResult> CreateChildContainer(
-            if (checkContainerInfo is not JsonResult<WopiCheckContainerInfo> jsonResult ||
-                jsonResult.Data is null)
+            var checkContainerInfo = await newFolder.GetWopiCheckContainerInfo(HttpContext);
+            return new JsonResult(
+                new CreateChildContainerResponse(
+                    new(newFolder.Name, Url.GetWopiSrc(WopiResourceType.Container, newFolder.Identifier)),
+                    checkContainerInfo));
+        }
+
+        return new InternalServerErrorResult();
+    }
+
+    /// <summary>
+    /// The CreateChildFile operation creates a new file in the provided container.
+    /// Specification: https://learn.microsoft.com/microsoft-365/cloud-storage-partner-program/rest/containers/createchildfile
+    /// Example URL path: /wopi/containers/(container_id)
+    /// </summary>
+    /// <param name="id">A string that specifies a container ID of a container managed by host. This string must be URL safe.</param>
+    /// <param name="suggestedTarget">A UTF-7 encoded string specifying either a file extension or a full file name, including the file extension</param>
+    /// <param name="relativeTarget">A UTF-7 encoded string that specifies a full file name including the file extension. The host must not modify the name to fulfill the request.</param>
+    /// <param name="overwriteRelativeTarget">A Boolean value that specifies whether the host must overwrite the file name if it exists. The default value is false.</param>
+    /// <param name="cancellationToken">cancellation token</param>
+    /// <returns></returns>
+    [HttpPost("{id}")]
+    [Produces(MediaTypeNames.Application.Json)]
+    [WopiOverrideHeader(WopiContainerOperations.CreateChildFile)]
+    [WopiAuthorize(WopiResourceType.Container, Permission.CreateChildFile)]
+    public async Task<IActionResult> CreateChildFile(
+        string id,
+        [FromHeader(Name = WopiHeaders.SUGGESTED_TARGET)] UtfString? suggestedTarget = null,
+        [FromHeader(Name = WopiHeaders.RELATIVE_TARGET)] UtfString? relativeTarget = null,
+        [FromHeader(Name = WopiHeaders.OVERWRITE_RELATIVE_TARGET)] bool? overwriteRelativeTarget = false,
+        CancellationToken cancellationToken = default)
+    {
+        if (writableStorageProvider is null)
+        {
+            return new NotImplementedResult();
+        }
+
+        var container = await storageProvider.GetWopiContainer(id, cancellationToken);
+        if (container is null)
+        {
+            return NotFound();
+        }
+
+        // the two headers are mutually exclusive. If both headers are present (or missing), the host should respond with a 501 Not Implemented status code.
+        if ((!string.IsNullOrWhiteSpace(suggestedTarget) && !string.IsNullOrWhiteSpace(relativeTarget)) ||
+            (string.IsNullOrWhiteSpace(suggestedTarget) && string.IsNullOrWhiteSpace(relativeTarget)))
```

Comment:
```suggestion
        if (string.IsNullOrWhiteSpace(suggestedTarget) ^ string.IsNullOrWhiteSpace(relativeTarget))
```

Reply:
that's not the same so it wont work...

Reply:
sorry...if you negate my condition, it should work, right?

Reply:
I don't want to block merging on this. let's take this offline